### PR TITLE
Fix issue #15555: Set `send_empty_value` for noWrapper.writeMetadata

### PR DESCRIPTION
**Pull request by ORabbit**
closes https://github.com/hashicorp/terraform-provider-google/issues/15555

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
pubsub: fixed issue where setting `no_wrapper.write_metadata` to false wasn't passed to the API for `google_pubsub_subscription`
```

### CODE REVIEW

**Comment by ORabbit** [comment]
> Waiting on acceptance tests. In the meantime is there an existing bug or issues associated with this ?
> 
> Concept seems fine

b/296582112 / https://github.com/hashicorp/terraform-provider-google/issues/15555

**Comment by ScottSuarez** [comment]
> Tests passed during RECORDING mode: `TestAccBigQueryDataTable_bigtable`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccBigQueryDataTable_bigtable.log)] `TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample.log)] `TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample.log)] `TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample.log)] `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccDataprocClusterIamPolicy.log)]
> 
> ##### Rerun these tests in REPLAYING mode to catch issues
> No issues found for passed tests after REPLAYING rerun.
> 
> Tests failed during RECORDING mode: `TestAccPubsubSubscription_pushNoWrapperEmpty`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/build-log/recording_build/TestAccPubsubSubscription_pushNoWrapperEmpty_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording/TestAccPubsubSubscription_pushNoWrapperEmpty.log)]
> 
> Please fix these to complete your PR. View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8739/artifacts/cd593b8c-57ae-4791-a77d-e48bdbea5fcd/recording) for each test



Looks like this is an API issue. We are setting the value and the API isn't returning it. We won't be able to resolve this without changes in the API.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Need API changes to support

**Comment by ScottSuarez** [comment]
Let me know if there is anything I can help with

**Comment by ScottSuarez** [comment]
Looks like a compilation issue, Let me know if you need any help

**Comment by ScottSuarez** [comment]
Could you pull the latest changes from main? Not clear why TGC is failing

**Comment by ScottSuarez** [comment]
panic crash from the VCR

```
panic: Invalid address to set: []string{"push_config", "0", "no_wrapper", "0", "writeMetadata"}
```

File: mmv1/templates/terraform/custom_flatten/pubsub_no_wrapper_write_metadata_flatten.go.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	if v == nil {
+		return nil
+	}
+
+	original := v.(map[string]interface{})
+	transformed := make(map[string]interface{})
+
+	if original["writeMetadata"] == nil {
+		transformed["write_metadata"] = false
+	}
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/custom_flatten/pubsub_no_wrapper_write_metadata_flatten.go.erb:25]:
```suggestion
	if original["writeMetadata"] == nil {
		transformed["write_metadata"] = false
	} else {
		transformed["write_metadata"] =  original["writeMetadata"]
	}
```

Looks like the value isn't getting set now in the case of true. This should fix it.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**